src/Hal/Serializer/CollectionNormalizer.php

```diff
@@ -68,15 +69,17 @@ public function normalize($object, $format = null, array $context = [])
         $resourceClass = $this->resourceClassResolver->getResourceClass($object, $context['resource_class'] ?? null, true);
         $context = $this->initContext($resourceClass, $context);
         $parsed = IriHelper::parseIri($context['request_uri'] ?? '/', $this->pageParameterName);
-        $paginated = $isPaginator = $object instanceof PaginatorInterface;
 
-        $currentPage = $lastPage = $itemsPerPage = null;
-        if ($isPaginator) {
+        $currentPage = $lastPage = $itemsPerPage = $pageTotalItems = null;
```

Comment:
Because of this?

```shell
------ --------------------------------------------- 
  Line   src/Hal/Serializer/CollectionNormalizer.php  
 ------ --------------------------------------------- 
  87     Undefined variable: $currentPage             
  92     Undefined variable: $lastPage                
  97     Undefined variable: $currentPage             
  98     Undefined variable: $currentPage             
  101    Undefined variable: $currentPage             
  101    Undefined variable: $pageTotalItems          
  101    Undefined variable: $itemsPerPage            
  102    Undefined variable: $currentPage             
  123    Undefined variable: $itemsPerPage            
 ------ --------------------------------------------- 
 ------ ---------------------------------------------------------- 
  Line   src/Hydra/Serializer/PartialCollectionViewNormalizer.php  
 ------ ---------------------------------------------------------- 
  75     Undefined variable: $currentPage                          
  80     Undefined variable: $lastPage                             
  85     Undefined variable: $currentPage                          
  86     Undefined variable: $currentPage                          
  89     Undefined variable: $currentPage                          
  89     Undefined variable: $pageTotalItems                       
  89     Undefined variable: $itemsPerPage                         
  90     Undefined variable: $currentPage                          
 ------ ---------------------------------------------------------- 
```